src/components/Navigation/Navigation.tsx

```diff
@@ -0,0 +1,262 @@
+              <div className="p-navigation__tagged-logo">
+                <a className="p-navigation__link" href={logo.url}>
+                  <div className="p-navigation__logo-tag">
+                    <img
+                      className="p-navigation__logo-icon"
+                      src={logo.src}
+                      alt=""
+                    />
+                  </div>
+                  <span className="p-navigation__logo-title">{logo.title}</span>
+                </a>
+              </div>
+            ) : (
+              <div className="p-navigation__logo">{logo}</div>
+            )
+          }
+          <ul className="p-navigation__items">
+            {
+              // When the header has a search box then this button is used to
+              // toggle the search box at mobile size.
+              hasSearch ? (
+                <li className="p-navigation__item">
+                  <button
+                    className="p-navigation__link--search-toggle"
+                    onClick={toggleSearch}
+                  >
+                    <span className="p-navigation__search-label">Search</span>
+                  </button>
+                </li>
+              ) : null
+            }
+            <li className="p-navigation__item">
+              <button
+                aria-pressed={mobileMenuOpen}
+                className="p-navigation__link"
+                onClick={toggleMobileMenu}
+              >
+                {mobileMenuOpen ? "Close menu" : "Menu"}
+              </button>
+            </li>
+          </ul>
+        </div>
+        <nav className="p-navigation__nav" {...navProps}>
+          {/* 
+            Always include the left nav list so that it takes up the space 
+            before the right nav list.
+          */}
+          <ul className="p-navigation__items" {...leftNavProps}>
+            {items ? generateItems(items, closeMobileMenu, generateLink) : null}
+          </ul>
+          {itemsRight || hasSearch ? (
+            <ul className="p-navigation__items" {...rightNavProps}>
+              {itemsRight
+                ? generateItems(itemsRight, closeMobileMenu, generateLink)
+                : null}
+              {
+                // When the header has a search box then this button is used to
+                // toggle the search box at non-mobile size.
+                hasSearch ? (
+                  <li className="p-navigation__item">
+                    <button
+                      className="p-navigation__link--search-toggle"
+                      onClick={toggleSearch}
+                    >
+                      <span className="p-navigation__search-label">Search</span>
+                    </button>
+                  </li>
+                ) : null
+              }
+            </ul>
+          ) : null}
+          {
+            // When the header has a search box and the user has opened the search
+            // form then this search box is displayed.
```

Comment:
QA note on the UX of this.

Do you think you could set the search input field to receive focus when activated on search toggle button?

This would remove 1 extra click that's really unnecessary here. Ideally a user should be able to start typing the search query right after clicking the search button.

Reply:
Thanks! The docs also mention to close when esc is pressed so I've added that too.